### ISSUE TITLE
Should IntegrityCheck match restoreConformity?

### ISSUE THREAD

**Issue by jandam** [gwlucastrig/Tinfour#27]
Method restoreConformity was modified to use delaunayThreshold instead of 0.0 to check condition for skipping restore conformity. But IntegrityCheck was not modified. I think that IntegrityCheck should use delaunayThreshold too. This issue is in both Standard & SemiVirtual implementations.

SemiVirtualIncrementalTin
https://github.com/gwlucastrig/Tinfour/blob/abae17d5cc8760686f503e4dda24e601a4dc4a51/src/main/java/tinfour/semivirtual/SemiVirtualIncrementalTin.java#L2118

SemiVirtualIntegrityCheck
https://github.com/gwlucastrig/Tinfour/blob/abae17d5cc8760686f503e4dda24e601a4dc4a51/src/main/java/tinfour/semivirtual/SemiVirtualIntegrityCheck.java#L343

**Comment by jandam** [comment]
Proposed fix starting at line 343 SemiVirtualIntegrityCheck:
```
    double h = geoOp.inCircle(a, b, c, d);

    if (h > 0) {
      if (e.isConstrained()) {
        // because the edge will not necessarily have had conformity
        // restored, we do not automatically treat the test as a failure here
        if (h > thresholds.getDelaunayThreshold()) {
          this.nDelaunayViolationsConstrained++;
          this.sumDelaunayViolationsConstrained += h;
          if (h > this.maxDelaunayViolationConstrained) {
            this.maxDelaunayViolationConstrained = h;
          }
        }
      } else {
        this.nDelaunayViolations++;
        this.sumDelaunayViolations += h;
        if (h > this.maxDelaunayViolation) {
          this.maxDelaunayViolation = h;
        }

        if (h > thresholds.getDelaunayThreshold()) {
          message = "InCircle failure h=" + h
            + ", starting at edge " + e
            + ": ("
            + a.getIndex()
            + ", " + b.getIndex()
            + ", " + c.getIndex()
            + ", " + d.getIndex() + ")";
          return false;

        }
      }
    }

```

**Comment by gwlucastrig** [comment]
You make a good observation.   The constrained-violation count was intended as a qualitative view of how often an inCircle() violation was suppressed due to a constrained edge.  It would be more meaningful if it were changed to match the logic in the restore-conformity modules.

Coincidentally, I recently changed the wording in the IntegrityCheck printSummary() method from:

- Detected Delaunay violations within tolerance

to read

- Detected acceptable Delaunay violations within tolerance

I think this better conveys the fact that the algorithm tolerates very-small violations for efficiency purposes.  Someday, when I have some spare time (as if that would ever happen), I want to try using Shewchuk's robust inCircle() implementation.  Shewchuk's predicates are supposed to always give the correct answer regardless of numerical precision issues.  It would be interesting to see how much performance overhead they require.
